stdlib/LinearAlgebra/test/blas.jl

```diff
@@ -641,6 +641,30 @@ end
     @test BLAS.get_num_threads() === default
 end
 
+@testset "with_num_threads" begin
+    prev_num_threads = BLAS.get_num_threads()
+    context_num_threads = BLAS.with_num_threads(1) do
+        BLAS.get_num_threads()
+    end
+    @test context_num_threads == 1
+    @test prev_num_threads == BLAS.get_num_threads()
+
+    @testset "thread unsafe" begin
+        prev_num_threads = BLAS.get_num_threads()
+        context_num_threads = 1
+        # task A
+        t = @async BLAS.with_num_threads(context_num_threads) do
+            sleep(0.5)
+        end
+        sleep(0.1)
```

Comment:
This execution order is not guaranteed. Use a barrier instead to guarantee the ordering of events